<!-- 真實世界驗證主要執行檔 -->

<launch>
  <!-- Arguments -->
  <arg name="scene"        default="0712"/>     <!-- 開啟地圖檔案 -->
  <arg name="open_rviz"    default="true"/>     <!-- 是否開啟 RViz -->
  <arg name="use_youbot"   default="true"/>     <!-- 使用 youBot (T) 或 Turtlebot3 Burger (F) -->
  <arg name="init_pos_x"   default="0.0"/>      <!-- 機器人模型初始 x -->
  <arg name="init_pos_y"   default="0.0"/>      <!-- 機器人模型初始 y -->
  <arg name="init_pos_z"   default="0.0"/>      <!-- 機器人模型初始 z -->
  <arg name="init_pos_yaw" default="0.0"/>      <!-- 機器人模型初始 yaw -->

  <!-- 發布地圖資訊 -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(find rrt_rl_nav)/maps/$(arg scene)/map.yaml"/>

  <!-- AMCL 定位資訊 -->
  <include file="$(find rrt_rl_nav)/launch/amcl.launch"/>

  <!-- 開啟Rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find rrt_rl_nav)/config/rrt_rl_navigation_rviz.rviz"/>
  </group>

</launch>
